### PR TITLE
latin64ble Keyboard ,newone with bluetooth4.0 module and like hhkb keyboard

### PR DESCRIPTION
**Pull request by latincompass**
latin64ble Keyboard,a new keyboard have RGB &Hot Swap

## Description

latin64ble Keyboard,a new keyboard have RGB &Hot Swap
## Types of Changes

- [x] Core
- [x] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/latin64ble/config.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+#define RGBLIGHT_ANIMATIONS
+
+#define RGBLIGHT_HUE_STEP 10
+#define RGBLIGHT_SAT_STEP 17
+#define RGBLIGHT_VAL_STEP 25
+#define RGBLIGHT_LIMIT_VAL 255
+#define RGBLIGHT_EFFECT_KNIGHT_OFFSET 20
+#define RGBLIGHT_ANIMATIONS             // Run RGB animations
```
[the remark this answers is not in the document]
Comment (latincompass) [reply keyboards/latin64ble/config.h:47]: I have revised it according to your opinion

File: keyboards/latin64ble/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#include QMK_KEYBOARD_H
+#define T1 MO(1)
+#define T2 MO(2)
+
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```
Comment (drashna) [inline keyboards/latin64ble/keymaps/via/keymap.c:7]: This also needs to have 4 layers, or set to limit it to 3 layers.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/latin64ble/keymaps/via/keymap.c (hunk 2 above)
Comment (latincompass) [reply keyboards/latin64ble/keymaps/via/keymap.c:7]: ihave fix this ,thanks a lot sir

**Review by drashna — APPROVED** [review]

File: keyboards/latin64ble/rules.mk [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+MCU = atmega32u4
+
+F_CPU = 8000000
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/latin64ble/rules.mk:11]:
```suggestion
# Processor frequency
F_CPU = 8000000

# Bootloader selection
```

File: keyboards/latin64ble/rules.mk [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+
+# Build Options
+#   change yes to no to disable
```
Comment (fauxpark) [inline keyboards/latin64ble/rules.mk:19]:
```suggestion
```

File: keyboards/latin64ble/rules.mk [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality on B7 by default
+MIDI_ENABLE = no            # MIDI controls
+UNICODE_ENABLE = no         # Unicode
+AUDIO_ENABLE = no           # Audio output on port C6
+RGBLIGHT_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/latin64ble/rules.mk:33]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/latin64ble/readme.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# Latin64ble
+
+![Latin64ble](https://github.com/latincompass/latin64BLE-kb)
```
Comment (fauxpark) [inline keyboards/latin64ble/readme.md:3]: This needs to be a proper image link.

File: keyboards/latin64ble/readme.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+macro 64 ble 4.0 RGB keyboard
+
+* Keyboard Maintainer: [latincompass](https://github.com/latincompass)
+* Hardware Availability: https://github.com/latincompass/latin64BLE-kb
+
+Make example for this keyboard (after setting up your build environment):
+
+    make latin64ble:default
+    make latin64ble:via
```
Comment (fauxpark) [inline keyboards/latin64ble/readme.md:13]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/latin64ble/readme.md (hunk 6 above)
Comment (latincompass) [reply keyboards/latin64ble/readme.md]: Thanks a lot,Sir

File: keyboards/latin64ble/readme.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+# Latin64ble
+
+![Latin64ble](https://github.com/latincompass/latin64BLE-kb/blob/main/latin64ble%20(1).JPG)
```
Comment (fauxpark) [inline keyboards/latin64ble/readme.md:3]:
```suggestion
![Latin64ble](https://i.imgur.com/zRlOU3ml.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**